drivers/platform/stm32/stm32_xspi.c

```diff
@@ -0,0 +1,718 @@
+/***************************************************************************//**
```

Comment:
so this is good in principle, I'd rename it and the function ops to stm32_octospi to reflect the true peripheral being used

and as I said in the other comment, make it use generic (and slightly modified) no_os_spi instead of a new no_os_xspi